transport/wsn/src/test/java/org/kaazing/gateway/transport/wsn/networkinterface/resolution/NetworkInterfaceResolutionIT.java

```diff
@@ -0,0 +1,99 @@
+/**
+ * Copyright 2007-2015, Kaazing Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kaazing.gateway.transport.wsn.networkinterface.resolution;
+import static org.kaazing.test.util.ITUtil.createRuleChain;
+
+import java.net.URI;
+
+import org.junit.Ignore;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestRule;
+import org.kaazing.gateway.server.test.GatewayRule;
+import org.kaazing.gateway.server.test.config.GatewayConfiguration;
+import org.kaazing.gateway.server.test.config.builder.GatewayConfigurationBuilder;
+import org.kaazing.k3po.junit.annotation.Specification;
+import org.kaazing.k3po.junit.rules.K3poRule;
+
+public class NetworkInterfaceResolutionIT {
+    
+    private GatewayRule gateway = new GatewayRule() {
+        {
+            // @formatter:off
+            GatewayConfiguration configuration =
+                    new GatewayConfigurationBuilder()
+                        .cluster()
+                        	//TODO: Add support for strings in cluster accepts
+                            .accept("tcp://[@Software Loopback Interface 1]:2345")
+                            .connect(URI.create("tcp://localhost:5432"))
+                            .name("clusterName")
+                        .done()
+                        .service()
+                            .accept("tcp://localhost:8003")
+                                .acceptOption("tcp.bind", "[@Software Loopback Interface 1]:7082")
+                            .connect(URI.create("tcp://localhost:8004"))
+                            .type("proxy")
+                        .done()
+//                        .service()
+//                            .accept("http://localhost:8110")
+//                                .acceptOption("http.transport", "tcp//[@Software Loopback Interface 1]:7081")
+//                            .connect(URI.create("http://localhost:8080"))
+//                            .type("proxy")
+//                        .done()
+                        .service()
+                            .accept("tcp://[@Software Loopback Interface 1]:8001")
```

Comment:
The current implementation for the gateway configuration builder relies on the fact that the parameters passed to the accept methods in both services and clustering contexts are URIs, e.g.:

>
GatewayConfiguration configuration =
                    new GatewayConfigurationBuilder()
                         .service()
                            .accept(URI.create("tcp://localhost:8003")) //URI passed as parameter
                            .connect(URI.create("tcp://localhost:8004"))
                            .type("proxy")
                        .done()

In order to ensure support for String \<accept\>s which do not comply to standard URIs (e.g. network interfaces syntax) the testing config builders needed extending with accept methods which can receive Strings as parameters, both in services and clustering contexts.

This seems a bit redundant to me, since we now have "accept"s stored as URIs and "stringAccept"s stored as Strings on both clusters and services. 

Normally, switching to Strings as parameters instead of URIs for accept methods would have seemed to be the better option to me, but this would have meant refactoring the entire testing framework, which I'm not sure would be the desired approach for now (Jitu, can you please confirm this?). 

I would have considered this to be the better approach also due to the fact that existing implementation involved building the URIs based on the strings passed as accept method parameters, and then in the org.kaazing.gateway.server.test.Gateway class, the default implementation performs the conversion back to Strings via the java.net.URI.toASCIIString() method (basically the existing implementation wraps a String in an URI and then performs the unwrap), e.g. (from org.kaazing.gateway.server.test.Gateway.java):

>
private void setCluster(GatewayConfig gatewayConfig, ClusterConfiguration cluster) {
...
Collection<URI> accepts = cluster.getAccepts();
...
for (String accept : stringAccepts) {
      newCluster.addAccept(accept);
}

Can you please provide your input on this?